lib/ansible/modules/network/dnac/dnac_rest.py

```diff
@@ -0,0 +1,238 @@
+        type: bool
+        default: 'yes'
+    content:
+        description:
+            - Raw content which should be fed in body.
+
+author:
+    - Kevin Breit (@kbreit)
+'''
+
+EXAMPLES = '''
+# Query inventory
+- name: Query network device inventory
+  sda_rest:
+    username: dnacuser
+    password: dnaccpass
+    host: dnac.yourorg.com
+    method: get
+    path: /api/v1/network-device
+    use_ssl: yes
+  delegate_to: localhost
+
+'''
+
+RETURN = '''
+response:
+    description: Data returned from controller.
+    type: string
+    returned: success
+version:
+    description: Version of returned data.
+    type: string
+    returned: info
+'''
+
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.urls import fetch_url, open_url
+from ansible.module_utils.basic import json
+from ansible.module_utils._text import to_native
+
+
+def main():
+
+    # define the available arguments/parameters that a user can pass to
+    # the module
+    module_args = dict(
+        username=dict(type='str', required=True),
+        password=dict(type='str', required=True, no_log=True),
+        host=dict(type='str', required=True),
+        method=dict(type='str', choices=['delete', 'get', 'post'], required=True),
+        path=dict(type='path', required=True),
+        timeout=dict(type='int', default=30),
+        use_proxy=dict(type='bool', default=False),
+        use_ssl=dict(type='bool', default=True),
+        validate_certs=dict(type='bool', default=True),
+        content=dict(type='raw'),
+    )
+
+    # seed the result dict in the object
+    # we primarily care about changed and state
+    # change is if this module effectively modified the target
+    # state will include any data that you want your module to pass back
+    # for consumption, for example, in a subsequent task
+    result = dict(
+        changed=False,
+    )
+
+    # the AnsibleModule object will be our abstraction working with Ansible
+    # this includes instantiation, a couple of common attr would be the
+    # args/params passed to the execution, as well as if the module
+    # supports check mode
+    module = AnsibleModule(
+        argument_spec=module_args,
+        supports_check_mode=False,
+    )
+
+    path = module.params['path']
+    content = module.params['content']
+    payload = content
+
+    # if the user is working with this module in only check mode we do not
+    # want to make any changes to the environment, just return the current
+    # state with no modifiupcations
+    if module.check_mode:
+        return result
+
+    # manipulate or modify the state as needed (this is going to be the
+    # part where your module will do what it needs to do)
+    protocol = None
+    if module.params['use_ssl'] is not None and module.params['use_ssl'] is False:
+        protocol = 'http'
+    else:
+        protocol = 'https'
+
+    url = '{0}://{1}{2}'.format(protocol, module.params['host'], module.params['path'])
```

Comment:
What if `path` doesn't start with a slash ?

Reply:
Revised in future commit.